pkg/apis/core/validation/validation.go

```diff
@@ -4458,13 +4523,31 @@ func validatePodConditions(conditions []core.PodCondition, fldPath *field.Path)
 	return allErrs
 }
 
+func CleanVolumeSourceList(podSpec *core.PodSpec) {
```

Comment:
Add a comment describing what this function does & how it's used.

(and in other places where you've added a function)

Reply:
done